File: fix_fonts.py
import re

with open('src/App.jsx', 'r') as f:
    content = f.read()

# For App.jsx homepage and work page, we'll replace the existing text sizes
# The user wants:
# Large -> text-2xl md:text-4xl
# Medium -> text-xl md:text-2xl
# Small -> text-[17px] md:text-[19px]

def replace_large(match):
    # For h1, h2, replacing text-3xl md:text-5xl or text-3xl with text-2xl md:text-4xl
    inner = match.group(2)
    inner = re.sub(r'text-\w+(-\w+)?(\smd:text-\w+(-\w+)?)?(\slg:text-\w+(-\w+)?)?', '', inner)
    inner = ' '.join(inner.split())
    inner += ' text-2xl md:text-4xl'
    return match.group(1) + inner + '"'

def replace_medium(match):
    # For h3, h4, replace with text-xl md:text-2xl
    inner = match.group(2)
    if 'tracking-widest' in inner and 'uppercase' in inner:
        # Keep labels small
        inner = re.sub(r'text-\w+(-\w+)?(\smd:text-\w+(-\w+)?)?', '', inner)
        inner = ' '.join(inner.split())
        inner += ' text-sm md:text-base'
    else:
        inner = re.sub(r'text-\w+(-\w+)?(\smd:text-\w+(-\w+)?)?', '', inner)
        inner = ' '.join(inner.split())
        inner += ' text-xl md:text-2xl'
    return match.group(1) + inner + '"'

def replace_small(match):
    # For p tags
    inner = match.group(2)
    if 'text-xs' in inner or 'text-[10px]' in inner or 'text-[12px]' in inner:
        return match.group(0) # Skip
    inner = re.sub(r'text-\w+(-\w+)?(\smd:text-\w+(-\w+)?)?', '', inner)
    inner = ' '.join(inner.split())
    inner += ' text-[17px] md:text-[19px]'
    return match.group(1) + inner + '"'

# Actually, doing this globally to App.jsx will be extremely intrusive and might break specific UI like forms or tooltips.
# The user specifically said "for the homepage work page everywhere".
# In App.jsx, I should just map the index.css variables!

File: fix_fonts.js
const fs = require('fs');

let content = fs.readFileSync('src/App.jsx', 'utf8');

// The 3 fonts:
// 1. Large Headers: font-primary text-5xl md:text-7xl lg:text-8xl
// 2. Sub Headers: font-primary text-xl md:text-2xl
// 3. Body: font-secondary text-[17px] md:text-[19px]

// We will replace existing sizes for h1, h2 with large headers
content = content.replace(/<h[12]([^>]*className="[^"]*)text-\d[a-z]+(\smd:text-\d[a-z]+)?(\slg:text-\d[a-z]+)?([^"]*)"/g, (match, p1, p2, p3, p4) => {
    // remove other font sizes
    let newClass = p1 + p4;
    return `<h2${newClass.replace(/text-\w+ /g, '')} font-primary text-5xl md:text-7xl lg:text-8xl"`;
});

// For h3, h4, we use the sub header size
content = content.replace(/<h[34]([^>]*className="[^"]*)text-\d[a-z]+(\smd:text-\d[a-z]+)?(\slg:text-\d[a-z]+)?([^"]*)"/g, (match, p1, p2, p3, p4) => {
    let newClass = p1 + p4;
    return `<h3${newClass.replace(/text-\w+ /g, '')} font-primary text-xl md:text-2xl"`;
});

// For paragraph text, replace with the body text font
content = content.replace(/<p([^>]*className="[^"]*)text-[a-z]+(\smd:text-[a-z]+)?([^"]*)"/g, (match, p1, p2, p3) => {
    let newClass = p1 + p3;
    return `<p${newClass.replace(/text-\w+ /g, '')} font-secondary text-[17px] md:text-[19px]"`;
});

// Fix t-display, t-body
content = content.replace(/t-display/g, "font-primary text-5xl md:text-7xl lg:text-8xl font-medium tracking-tight");
content = content.replace(/t-body/g, "font-secondary text-[17px] md:text-[19px] leading-relaxed font-normal");
content = content.replace(/t-label/g, "font-primary text-sm md:text-base tracking-widest uppercase font-bold");

fs.writeFileSync('src/App.jsx.modified', content);
